### PR TITLE
fix query latest salt changes when the bundle is used

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

fix query latest salt changes when the bundle is used

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: testsuite/features/step_definitions/command_steps.rb [hunk 1]
```diff
@@ -180,7 +180,11 @@
 
 When(/^I query latest Salt changes on "(.*?)"$/) do |host|
   node = get_target(host)
-  result, return_code = node.run("LANG=en_US.UTF-8 rpm -q --changelog salt")
+  salt = 'salt'
+  if host != 'server' && host != 'proxy'
+    salt = $product == 'Uyuni' ? "venv-salt-minion" : "salt"
+  end
+  result, return_code = node.run("LANG=en_US.UTF-8 rpm -q --changelog #{salt}")
```
Comment (vzhestkov) [inline testsuite/features/step_definitions/command_steps.rb:187]: @mcalmer maybe just modify the command this way?
```
rpm -q --changelog venv-salt-minion || rpm -q --changelog salt
```

**Review by srbarrios — APPROVED** [review]

File: testsuite/features/step_definitions/command_steps.rb (hunk 1 above)
Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:187]: I prefer Michael's more deterministic approach.

**Review by ktsamis — APPROVED** [review]

File: testsuite/features/step_definitions/command_steps.rb (hunk 1 above)
Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:187]: As we are distinguished by product version in the rest of the steps... better do it also here.

**Review by Bischoff — APPROVED** [review]

File: testsuite/features/step_definitions/command_steps.rb (hunk 1 above)
Comment (vzhestkov) [reply testsuite/features/step_definitions/command_steps.rb:187]: ok

**Review by vzhestkov — APPROVED** [review]

File: testsuite/features/step_definitions/command_steps.rb [hunk 2]
```diff
@@ -180,7 +180,11 @@
 
 When(/^I query latest Salt changes on "(.*?)"$/) do |host|
   node = get_target(host)
-  result, return_code = node.run("LANG=en_US.UTF-8 rpm -q --changelog salt")
+  salt = 'salt'
+  if host != 'server' && host != 'proxy'
```
Comment (srbarrios) [inline testsuite/features/step_definitions/command_steps.rb:184]:
```suggestion
  if host != 'server'
```

Comment (mcalmer) [reply testsuite/features/step_definitions/command_steps.rb]: I did it different as I do not know what rubocop will say about this

File: testsuite/features/step_definitions/command_steps.rb [hunk 3]
```diff
@@ -189,7 +193,8 @@
 
 When(/^I query latest Salt changes on ubuntu system "(.*?)"$/) do |host|
   node = get_target(host)
-  result, return_code = node.run("zcat /usr/share/doc/salt-minion/changelog.Debian.gz")
+  salt = $product == 'Uyuni' ? "venv-salt-minion" : "salt-minion"
+  result, return_code = node.run("zcat /usr/share/doc/#{salt}/changelog.Debian.gz")
```
Comment (meaksh) [inline testsuite/features/step_definitions/command_steps.rb:197]: In of Debian, the changelog file for `venv-salt-minion` package is slightly different: named `changelog.gz`

```suggestion
  changelog_file = $product == 'Uyuni' ? "changelog.gz" : "changelog.Debian.gz"
  result, return_code = node.run("zcat /usr/share/doc/#{salt}/#{changelog_file}")
```

**Review by meaksh — COMMENTED** [review]
Thanks for the fix @mcalmer . I have also suggested a small change that it is also required

**Review by meaksh — APPROVED** [review]

**Pull request merged**